sdk/nodejs/tests/runtime/langhost/cases/015.runtime_sxs_mocks/index.js

```diff
@@ -0,0 +1,43 @@
+let assert = require("assert");
+let path = require("path");
+
+const sdkPath = "../../../../../";
+
+// Load the first copy:
+let pulumi1 = require(sdkPath);
+
+// Now delete the entries in the require cache, and load up the second copy:
+const resolvedSdkPath = path.dirname(require.resolve(sdkPath));
+Object.keys(require.cache).forEach((path) => {
+    if (path.startsWith(resolvedSdkPath)) {
+        delete require.cache[path];
+    }
+});
+let pulumi2 = require(sdkPath);
+
+// Make sure they are different:
+assert(pulumi1 !== pulumi2, "pulumi1 !== pulumi2");
+assert(pulumi1.runtime !== pulumi2.runtime, "pulumi1.runtime !== pulumi2.runtime");
+
+// Check that we can set mocks successfully
+// We don't need an actual test monitor here, just something to set and get.
+class Mocks {
+    newResource(args) {
+        return {
+            id: args.inputs.name + "_id",
+            state: {
+                ...args.inputs,
+            },
+        };
+    }
+    call(args) {
+        return args;
+    }
+}
+
+pulumi1.runtime.setMocks(new Mocks());
+assert(pulumi1.runtime.getMonitor() !== undefined);
+assert(pulumi2.runtime.getMonitor() !== undefined);
+assert.strictEqual(pulumi1.runtime.getMonitor(), pulumi2.runtime.getMonitor());
+assert(pulumi1.runtime.hasMonitor());
+assert(pulumi2.runtime.hasMonitor());
```

Comment:
This test breaks somehow, throwing an exception:

```
log: Running program '/home/tgummerer/work/pulumi/set-mock-monitor-in-global-store/sdk/nodejs/bin/tests/runtime/langhost/cases/015.runtime_sxs_mocks/' failed with an unhandled exception:
<ref *1> Error: unknown resource 
    at MockMonitor.registerResourceOutputs (/home/tgummerer/work/pulumi/set-mock-monitor-in-global-store/sdk/nodejs/bin/runtime/mocks.js:15:7656)
    at /home/tgummerer/work/pulumi/set-mock-monitor-in-global-store/sdk/nodejs/runtime/resource.ts:309:57
    at new Promise (<anonymous>)
    at Object.<anonymous> (/home/tgummerer/work/pulumi/set-mock-monitor-in-global-store/sdk/nodejs/runtime/resource.ts:309:57)
    at Generator.next (<anonymous>)
    at fulfilled (/home/tgummerer/work/pulumi/set-mock-monitor-in-global-store/sdk/nodejs/bin/runtime/resource.js:15:808) {
  promise: Promise { <rejected> [Circular *1] }
}
```

I'm not entirely sure why we even call registerResourceOutputs in the end here, since we're not exporting anything.  I started at this entirely too long already, so if someone has an idea of how to fix this, I'd really appreciate it.  Setting the monitor for both of the SDKs seems to work fine.

Reply:
Is this fixed now? I guess the issue was that there were 2 of the "side-by-side" tests that fiddled with the global module cache?

Reply:
The mocks interact in the same way with this test without the rest of my changes, so I don't think there's anything new broken here.  I'm still not 100% sure why this is happening, but the fix in this PR should still be valid.